tests/basic.cpp

```diff
@@ -1,21 +1,72 @@
 #include <ncrypto.h>
 
 #include <gtest/gtest.h>
-
 #include <string>
-#include <unordered_set>
 
 using namespace ncrypto;
 
-TEST(basic, cipher_foreach) {
-  std::unordered_set<std::string> foundCiphers;
+// Convenience class for creating buffers in tests
+struct TestBuf : public std::string {
+  TestBuf(const std::string& constStr)
+      : std::string(constStr),
+        buf{reinterpret_cast<unsigned char*>(data()), size()} {}
+  TestBuf(size_t n) : TestBuf(std::string(n, 0)) {}
+
+  operator Buffer<unsigned char>&() { return buf; }
+
+  Buffer<const unsigned char> asConst() const {
+    return Buffer<const unsigned char>{
+        .data = reinterpret_cast<const unsigned char*>(data()), .len = size()};
+  }
+
+ private:
+  Buffer<unsigned char> buf;
+};
+
+TEST(basic, test_it) {
+  SUCCEED();
+}
+
+#ifdef OPENSSL_IS_BORINGSSL
```

Comment:
This defeats the purpose of ncrypto isn't it? I don't see any tests validating the interface that doesn't distinguish between boringssl and openssl

Reply:
Indeed, this interface is Boring-specific. There turns out to be a huge chasm in how OpenSSL and BoringSSL want you to work with AEADs. I suspect it's possible to come back to this and make the AEAD class work with OpenSSL, but shoehorning Boring's approach into Cipher would have been very ugly.